_posts/2022-10-19-quarkus-observability-roadmap-2023.adoc

```diff
@@ -0,0 +1,90 @@
+https://opentracing.io[OpenTracing] and http://opencensus.io/[OpenCensus] to form
+http://opentelemetry.io[OpenTelemetry].
+
+Quarkus released the first version of the OpenTelemetry Extension with Quarkus 1.13.0.Final, in Q1
+of 2021. The initial support was only for HTTP Tracing (Vert.x and REST) but slowly moved to other components,
+including CDI, gRPC, Reactive Messaging, and many more, while also adopting and adjusting to all the new APIs released
+by OpenTelemetry Tracing. Our goal is to continue this work, expand OpenTelemetry Tracing to the remaining components
+and close the integration gaps between Quarkus and Tracing.
+
+*Moving forward, https://github.com/smallrye/smallrye-opentracing/[SmallRye OpenTracing] and all the
+https://opentracing.io[OpenTracing] work in Quarkus are now deprecated!*
+
+While things will likely continue to work, we encourage all our
+users to move to http://opentelemetry.io[OpenTelemetry], for all the Tracing capabilities. In the following upcoming
+versions, we may start to remove https://opentracing.io[OpenTracing] support from some extensions and continue that
+work until we completely remove https://opentracing.io[OpenTracing] from https://quarkus.io[Quarkus].
+
+For http://microprofile.io[MicroProfile] users out there that rely on
+https://github.com/eclipse/microprofile-opentracing[MicroProfile OpenTracing], MicroProfile 6.0, which will be
+supported by Quarkus 3 (and the Jakarta rename), ships
+https://github.com/eclipse/microprofile-telemetry[MicroProfile Telemetry Tracing]. This new specification defines the
+behaviors that allow MicroProfile applications to participate in an environment where distributed tracing is enabled
+via OpenTelemetry Tracing.
+
+== Metrics and Micrometer
+
+On the Metrics front, it became evident that while https://github.com/eclipse/microprofile-metrics[MicroProfile Metrics]
+and https://github.com/smallrye/smallrye-metrics/[SmallRye Metrics] were acceptable for simple
+use cases, they had some issues with more advanced ones. It also didn't help that users from different technologies had
+to learn new APIs and constructs to extract metrics information from their applications. We made a move to adopt
+https://micrometer.io[Micrometer] to resolve some of these issues while at the same time offering a less intrusive
+way to monitor Quarkus applications.
+
+We know that a new version of https://github.com/eclipse/microprofile-metrics[MicroProfile Metrics] is in the works,
+and an implementation provided by https://github.com/smallrye/smallrye-metrics/[SmallRye Metrics] will be available.
+Unfortunately, at this time, we feel that the current
+https://github.com/eclipse/microprofile-metrics[MicroProfile Metrics] path does not address many of our concerns.
+
+We intend to keep the https://micrometer.io[Micrometer] approach and expand it to the remaining areas. With
+OpenTelemetry Metrics reaching a stable milestone and Micrometer compromise to support
```

Comment:
There is no compromise here.. it was always the intent (from the beginning). Language suggests this was a concession of some kind...

Reply:
It was in the context of Micrometer taking the step also to support OpenTelemetry (they could have decided not to support it).

Do you have a different suggestion?